gcpbuildcache/src/main/kotlin/androidx/build/gradle/gcpbuildcache/GcpStorageService.kt

```diff
@@ -122,18 +121,27 @@ internal class GcpStorageService(
         }
 
         private fun credentials(
-            path: File,
+            gcpCredentials: GcpCredentials,
             isPushSupported: Boolean
         ): GoogleCredentials? {
-            if (!path.exists()) throw GradleException("Your specified $path does not exist")
-            if (!path.isFile) throw GradleException("Your specified $path is not a file")
             val scopes = mutableListOf(
                 STORAGE_READ_ONLY,
             )
             if (isPushSupported) {
                 scopes += listOf(STORAGE_READ_WRITE, STORAGE_FULL_CONTROL)
             }
-            return GoogleCredentials.fromStream(path.inputStream()).createScoped(scopes)
+            return when (gcpCredentials) {
+                is ApplicationDefaultGcpCredentials -> {
+                    GoogleCredentials.getApplicationDefault().createScoped(scopes)
+                }
+                is ExportedKeyGcpCredentials -> {
+                    val path = gcpCredentials.pathToCredentials
+                    if (!path.exists()) throw GradleException("Your specified $path does not exist")
```

Comment:
I don't remember seeing "Your" in an error message before - maybe that's because the user viewing the error might be different from the user that configured the path? Maybe "Credentials path $path does not exist"?

Reply:
Fixed.